man/CMakeLists.txt

```diff
@@ -11,12 +11,35 @@ if (WITH_MAN)
 		${CMAKE_BINARY_DIR}/libiio.3.in
 		${CMAKE_BINARY_DIR}/man/libiio.3 @ONLY)
 
+	if (WITH_DOC)
+		find_program(MAN2HTML man2html)
+		if (NOT MAN2HTML)
+			message(FATAL_ERROR "Can not build html DOC from man without man2html")
+		endif()
+		message(STATUS "Building html doc pages with man2html")
+		file(MAKE_DIRECTORY ${CMAKE_HTML_DEST_DIR}/man3)
+		execute_process(
+			COMMAND ${MAN2HTML} -r
+			INPUT_FILE ${CMAKE_BINARY_DIR}/man/libiio.3
+			OUTPUT_FILE ${CMAKE_HTML_DEST_DIR}/man3/libiio.3.html
+		)
+	endif()
 	if (WITH_TESTS)
-		configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/iio_attr.1.in ${CMAKE_BINARY_DIR}/man/iio_attr.1 @ONLY)
-		configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/iio_info.1.in ${CMAKE_BINARY_DIR}/man/iio_info.1 @ONLY)
-		configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/iio_readdev.1.in ${CMAKE_BINARY_DIR}/man/iio_readdev.1 @ONLY)
-		configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/iio_reg.1.in ${CMAKE_BINARY_DIR}/man/iio_reg.1 @ONLY)
-		configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/iio_writedev.1.in ${CMAKE_BINARY_DIR}/man/iio_writedev.1 @ONLY)
+		list(APPEND MAN1 iio_attr.1 iio_info.1 iio_readdev.1 iio_reg.1 iio_writedev.1)
+		foreach(_page ${MAN1})
+			configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/${_page}.in
+				${CMAKE_BINARY_DIR}/man/${_page} @ONLY)
+		endforeach()
+		if (WITH_DOC)
+			file(MAKE_DIRECTORY ${CMAKE_HTML_DEST_DIR}/man1)
+			foreach(_page ${MAN1})
+				execute_process(
+					COMMAND ${MAN2HTML} -r
+					INPUT_FILE ${CMAKE_BINARY_DIR}/man/${_page}
+					OUTPUT_FILE ${CMAKE_HTML_DEST_DIR}/man1/${_page}.html
+				)
+			endforeach()
+		endif()
```

Comment:
following a bit on comment https://github.com/analogdevicesinc/libiio/pull/413/commits/963b5e4e2b000cf0eb6a1913be1bade5be0d4e0f#r407344061  [from the formatting PR] ; maybe we should also resolve the aspect of `endif()` vs `endif__space__()`

Reply:
I figured I would fix all the things I had planned, and then go back and fix the formatting issues. (which we might have to fix some style issues with codacy now, similar to the recent python fixes).